Update the firware binary to work with latest cc3200-rs

fabricedesre opened the pull request:
Changes in this PR are:
- Update to current cc3200-rs master
- src/main.rs is now a simple runner, right now sending temperature readings.
- the wlan setup moved to wlan.rs
- config.rs.sample needs to be copied to config.rs with your actual parameters.

fabricedesre commented:
ping @dhylands 
Travis seems to have a hard time with cargo right now...

dhylands commented on src/config.rs.sample:
```diff
@@ -0,0 +1,17 @@
+}
+
+pub const SENSOR_READING_COUNT: u32 = 10;
+pub const SERVER_URL: &'static str = "http://10.252.33.211:8000/endpoint";
```
nit: no newline at end of file

dhylands commented on scripts/travis.sh:
```diff
@@ -9,5 +9,6 @@ export PATH=/home/travis/.cargo/bin:$PATH
 rustup default nightly
 rustup component add rust-src
 cargo install xargo
+cp src/config.rs.sample src/config.rs
```
That seems like a reasonable way to do it in cc3200-rs as well (and addresses your concern about checking the file in accidentally).

dhylands commented on src/main.rs:
```diff
@@ -9,17 +9,72 @@
+
+    for _ in 0..config::SENSOR_READING_COUNT {
+        let temperature: u32 = (temp_sensor.get_temperature().unwrap() * 10.0) as u32;
+        info!("Feels like {}.{} C", temperature / 10, temperature % 10);
```
nit: This will report interesting results with a negative temperature :)

dhylands commented on src/main.rs:
```diff
@@ -9,17 +9,72 @@
+        info!("Received {}",
+              response.body.read_string_to_end(&mut buffer).unwrap());
+        CurrentTask::delay(Duration::ms(1000))
+    }
```
I don't see where this actually sends a temperature to the server.

fabricedesre commented:
@dhylands I addressed comments and pinned Rustc to the same version as in cc3200-rs. That fixed the unrelated cargo issue which is https://github.com/rust-lang/cargo/issues/3340

dhylands commented:
lgtm
